python/cuspatial/cuspatial/core/binpreds/feature_touches.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from cuspatial.core.binpreds.binpred_interface import NotImplementedPredicate
+from cuspatial.core.binpreds.feature_contains import (
+    ContainsPredicateBase,
+    PointPointContains,
+)
+from cuspatial.utils.binpred_utils import (
+    LineString,
+    MultiPoint,
+    Point,
+    Polygon,
+    _false_series,
+)
+
+
+class TouchesPredicateBase(ContainsPredicateBase):
+    """Base class for binary predicates that use the contains predicate
+    to implement the touches predicate. For example, a Point-Polygon
+    Touches predicate is defined in terms of a Point-Polygon Contains
+    predicate.
+
+    Used by:
+    (Point, Polygon)
+    (Polygon, Point)
+    (Polygon, MultiPoint)
+    (Polygon, LineString)
+    (Polygon, Polygon)
+    """
+
+    pass
+
+
+class PointPointTouches(PointPointContains):
+    """Points can't touch according to GeoPandas, so return False."""
+
+    def _preprocess(self, lhs, rhs):
+        return _false_series(len(lhs))
+
+
+DispatchDict = {
+    (Point, Point): PointPointTouches,
```

Comment:
If points can't touch, shouldn't you dispatch to Impossible instead of a custom implementation?

Reply:
Yes, thank you.